RInAppMessaging/Classes/Utilities/AtomicWrapper.swift

```diff
@@ -15,12 +17,12 @@ struct AtomicGetSet<Value> {
             return queue.sync { value }
         }
         set {
-            queue.sync { value = newValue }
+            queue.sync(flags: .barrier) { value = newValue }
```

Comment:
changing `AtomicGetSet` to class and using concurrent queue fixes intermittent crashes that were happening (I don't fully understand why though)

Reply:
Do you have a stacktrace of the crash(es)? It looks like you've seen [Donny Wals' suggestions](https://www.donnywals.com/why-your-atomic-property-wrapper-doesnt-work-for-collection-types/) to fix the problem. But it still seems our implementation might not do so well with concurrency and collection value-types especially when collection value types are copied by each thread and copies also given out in the property wrapper's `wrappedValue.get` like he mentions

Reply:
@apparition47 That's why I tried to combine his AtomicDict and property wrapper which resulting in changing struct to class and adding concurrent queue. It seems like it helps but to be honest I'm not sure what is what at this point. (I'd have to calmly analyse this case one more time).
Here's the stacktrace (crashed on current line):
```
* thread #20, queue = 'IAM.AtomicProperty489)', stop reason = EXC_BAD_ACCESS (code=1, address=0x3be9ea32eab0)
    frame #0: 0x000000010924f2bd libswiftCore.dylib`_swift_release_dealloc + 13
    frame #1: 0x0000000108c0a0bd RInAppMessaging`partial apply for specialized closure #1 in AtomicGetSet.wrappedValue.setter [inlined] generic not re-abstracted specialization <Swift.Dictionary<Swift.String, RInAppMessaging.UserDataCacheContainer>> of closure #1 () -> () in RInAppMessaging.AtomicGetSet.wrappedValue.setter : τ_0_0 at <compiler-generated>:0 [opt]
    frame #2: 0x0000000108c0a09e RInAppMessaging`partial apply for specialized closure #1 in AtomicGetSet.wrappedValue.setter at <compiler-generated>:0 [opt]
    frame #3: 0x0000000108be5dcc RInAppMessaging`thunk for @callee_guaranteed () -> () at <compiler-generated>:0 [opt]
    frame #4: 0x0000000108be5dee RInAppMessaging`thunk for @escaping @callee_guaranteed () -> () at <compiler-generated>:0 [opt]
    frame #5: 0x00007fff201078df libdispatch.dylib`_dispatch_client_callout + 8
    frame #6: 0x00007fff201155e1 libdispatch.dylib`_dispatch_lane_barrier_sync_invoke_and_complete + 94
  * frame #7: 0x0000000108c08e97 RInAppMessaging`specialized AtomicGetSet.wrappedValue.setter(newValue=<unavailable>, self=<unavailable>) at AtomicWrapper.swift:20:19 [opt]
    frame #8: 0x0000000108c62caf RInAppMessaging`UserDataCache.cacheCampaignData(data=0 values, userIdentifiers=<unavailable>, self=0x0000600000786980) at UserDataCache.swift:55:36 [opt]
    frame #9: 0x0000000109e97d94 Tests`merged closure #1 () -> () in closure #1 () -> () in closure #8 () -> () in closure #2 () -> () in Tests.UserDataCacheSpec.spec() -> () + 212
    frame #10: 0x0000000109e97c7e Tests`thunk for @escaping @callee_guaranteed () -> () at <compiler-generated>:0 [opt]
    frame #11: 0x00007fff2010670d libdispatch.dylib`_dispatch_call_block_and_release + 12
    frame #12: 0x00007fff201078df libdispatch.dylib`_dispatch_client_callout + 8
    frame #13: 0x00007fff2010de15 libdispatch.dylib`_dispatch_lane_serial_drain + 715
    frame #14: 0x00007fff2010e98c libdispatch.dylib`_dispatch_lane_invoke + 400
    frame #15: 0x00007fff20118f81 libdispatch.dylib`_dispatch_workloop_worker_thread + 772
    frame #16: 0x00007fff6034045d libsystem_pthread.dylib`_pthread_wqthread + 314
    frame #17: 0x00007fff6033f42f libsystem_pthread.dylib`start_wqthread + 15
```

Reply:
@imaginaris I'd like this change considered separately. Please extract the AtomicWrapper related changes to another PR and ensure there is a JIRA tracking it. 

I think this change before/after is a good candidate for the automated stress tests we are planning.

Reply:
I think that this change should also be done in RAnalytics too as it uses AtomicGetSet.